### ISSUE TITLE
vscode not detecting multiple tsconfig files properly

### ISSUE THREAD

**Issue by danikaze** [microsoft/vscode#100432]
Possible bug report:

- VSCode Version: 1.46.0
- OS Version: Windows 10

More details:
```
Version: 1.46.0 (system setup)
Commit: a5d1cc28bb5da32ec67e86cc50f84c67cc690321
Date: 2020-06-10T09:03:20.462Z
Electron: 7.3.1
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18362
```

Steps to Reproduce:

1. Clone the [prepared repository](https://github.com/danikaze/vscode-multi-tsconfig) to reproduce the issue
2. Open [renderer/pages/home.tsx](https://github.com/danikaze/vscode-multi-tsconfig/blob/master/renderer/pages/home.tsx)
3. vscode shows the error `Cannot find name 'PACKAGE_NAME'.ts(2304)` on line 12

It should detect the definition from [renderer/td/env.d.ts](https://github.com/danikaze/vscode-multi-tsconfig/blob/master/renderer/td/env.d.ts) -- included by [renderer/tsconfig.json](https://github.com/danikaze/vscode-multi-tsconfig/blob/master/renderer/tsconfig.json)

Details are also described in the [README.md](https://github.com/danikaze/vscode-multi-tsconfig#readme) of the repo

Still occurs even running it with `code --disable-extensions`.

**Comment by mjbvz** [comment]
Thanks for the example project

The issue is that the `pages/home` is not part of any tsconfig. You can check this by opening the file and running: `TypeScript: go to project configuration`. 

The reason it is not included is this line:

```
  "include": ["./td/*.d.ts"]
```

Once I get rid of that include restrictions, the symbol is properly resolved

**Comment by danikaze** [comment]
Shouldn't `pages/home` be included in the root [tsconfig.json](https://github.com/danikaze/vscode-multi-tsconfig/blob/master/tsconfig.json)?
```
  "include": [
    "renderer/next-env.d.ts",
    "**/*.ts",
    "**/*.tsx"
  ],
```

**Comment by mjbvz** [comment]
The `include` in the more specific `tsconfig` overrides the `include` from the `tsconfig` it is extending

**Comment by danikaze** [comment]
OK! I got to properly reproduce the error which I was facing in my _real_ repository (not the mini one to reproduce it)

I added a new [commit](https://github.com/danikaze/vscode-multi-tsconfig/commit/5167a4f99cf222de61f3653df8792ee74ee70dc0) to that one so it's easily testable...
Maybe not a bug, but a feature I don't know... but allow me to ask in this thread if you don't mind.

So, once I got the [env.d.ts](https://github.com/danikaze/vscode-multi-tsconfig/blob/master/renderer/td/env.d.ts) working with your advice, adding the next step breaks it.
Basically, if in that file I add an `import`... all the existing declarations are not valid anymore for vscode (project still works)
